### PR TITLE
[Merged by Bors] - feat: the category of presheaves of modules is preadditive

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Category/ModuleCat/Presheaf.lean [hunk 1]
```diff
@@ -106,27 +106,100 @@ instance : Category (PresheafOfModules R) where
...
+  simp only [Limits.zero_app, AddMonoidHom.zero_apply, smul_zero])⟩
+
+variable (P Q)
+
+@[simp]
+lemma zero_app (X : Cᵒᵖ) : (0 : P ⟶ Q).app X = 0 := rfl
+
+variable {P Q}
+
+instance : Add (P ⟶ Q) := ⟨fun f g => Hom.mk (f.hom + g.hom) (by
```
Comment (TwoFX) [inline Mathlib/Algebra/Category/ModuleCat/Presheaf.lean:142]: Optional suggestion, you can decide which way you prefer.
```suggestion
instance : Add (P ⟶ Q) := ⟨fun f g => mk (f.hom + g.hom) (by
```

File: Mathlib/Algebra/Category/ModuleCat/Presheaf.lean [hunk 2]
```diff
@@ -106,27 +106,100 @@ instance : Category (PresheafOfModules R) where
...
+  intros
+  simp only [NatTrans.app_add, AddCommGroupCat.hom_add_apply, map_smul, smul_add])⟩
+
+@[simp]
+lemma add_app (f g : P ⟶ Q) (X : Cᵒᵖ) : (f + g).app X = f.app X + g.app X := rfl
+
+instance : Sub (P ⟶ Q) := ⟨fun f g => Hom.mk (f.hom - g.hom) (by
```
Comment (TwoFX) [inline Mathlib/Algebra/Category/ModuleCat/Presheaf.lean:149]: See above
```suggestion
instance : Sub (P ⟶ Q) := ⟨fun f g => mk (f.hom - g.hom) (by
```

File: Mathlib/Algebra/Category/ModuleCat/Presheaf.lean [hunk 3]
```diff
@@ -106,27 +106,100 @@ instance : Category (PresheafOfModules R) where
...
+  intros
+  rw [NatTrans.app_sub, AddMonoidHom.sub_apply, AddMonoidHom.sub_apply,
+    smul_sub, map_smul, map_smul])⟩
+
+@[simp]
+lemma sub_app (f g : P ⟶ Q) (X : Cᵒᵖ) : (f - g).app X = f.app X - g.app X := rfl
+
+instance : Neg (P ⟶ Q) := ⟨fun f => Hom.mk (-f.hom) (by
```
Comment (TwoFX) [inline Mathlib/Algebra/Category/ModuleCat/Presheaf.lean:157]: See above
```suggestion
instance : Neg (P ⟶ Q) := ⟨fun f => mk (-f.hom) (by
```

File: Mathlib/Algebra/Category/ModuleCat/Presheaf.lean [hunk 4]
```diff
@@ -106,27 +106,100 @@ instance : Category (PresheafOfModules R) where
...
+
 @[ext]
-theorem Hom.ext {f g : P ⟶ Q} (w : ∀ X, f.app X = g.app X) : f = g := by
+theorem ext {f g : P ⟶ Q} (w : ∀ X, f.app X = g.app X) : f = g := by
   cases f; cases g;
   congr
   ext X x
   exact LinearMap.congr_fun (w X) x
 
+instance : Zero (P ⟶ Q) := ⟨Hom.mk 0 (by
```
Comment (TwoFX) [inline Mathlib/Algebra/Category/ModuleCat/Presheaf.lean:131]: See above
```suggestion
instance : Zero (P ⟶ Q) := ⟨mk 0 (by
```

**Review by TwoFX — APPROVED** [review]
Thanks!
bors d+

**Comment by joelriou** [comment]
Thanks for the review!

bors merge

**Pull request closed without merging**